services/gen.go

```diff
@@ -2,3 +2,12 @@ package services
 
 //go:generate mockgen -destination=./mocks/pubkeycompendium.go -package=mocks github.com/Layr-Labs/eigensdk-go/services/pubkeycompendium PubkeyCompendiumService
 //go:generate mockgen -destination=./mocks/avsregistry.go -package=mocks github.com/Layr-Labs/eigensdk-go/services/avsregistry AvsRegistryService
+
+// We generate it in ./mocks/blsagg/ instead of ./mocks like the others because otherwise we get a circular dependency
+// avsregistry -> mocks -> avsregistry
+// because avsregistry_chaincaller_test -> for pubkeycompendium mock
+// and blsaggregation mock -> avsregistry interface
+// TODO: are there better ways to organize these dependencies? Maybe by using ben johnson
```

Comment:
yea might be a better idea to put mocks in the packages itself then in the uber mocks in`services`

Reply:
We can clean this later.